packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/math_expr.rs

```diff
@@ -0,0 +1,273 @@
+    pub fn add(&self, term: MathOrPrimitive) -> MathExpr {
+        let term = serde_json::to_string(&term).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).add(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, term
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a new mathematical expression by subtracting `term` to the current expression.
+    pub fn subtract(&self, term: MathOrPrimitive) -> MathExpr {
+        let term = serde_json::to_string(&term).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).subtract(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, term
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a new mathematical expression by multiplying the current expression by `term`.
+    pub fn multiply(&self, factor: MathOrPrimitive) -> MathExpr {
+        let factor = serde_json::to_string(&factor).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).multiply(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, factor
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a new mathematical expression by dividing the current expression by `term`.
+    pub fn divide(&self, factor: MathOrPrimitive) -> MathExpr {
+        let factor = serde_json::to_string(&factor).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).divide(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, factor
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+}
+
+/// An enum of different types of values that can be interpreted as mathematical expressions.
+#[derive(Debug, serde::Serialize)]
+#[serde(untagged)]
+pub enum MathOrPrimitive {
+    Math(MathExpr),
+    Number(f64),
+    Integer(i64),
+    String(String),
+}
+
+/// Parameters for creating a latex string from a `Math`:
+#[derive(Debug)]
+pub struct ToLatexParams {
+    /// If present, then pad numbers with zeros so they have at least
+    /// this many decimal places after the decimal point displayed.
+    pub pad_to_decimals: Option<i32>,
```

Comment:
Negatives are allowed? `u32`?

Reply:
Right, we can round to a negative number of decimals, but we can't pad to a negative number.